### ISSUE TITLE
Console reporter double reports

### ISSUE THREAD

**Issue by pJunger** [philsquared/Catch#924]
#### Problem:
When testing with the console reporter and nested checks, the reporter double-reports:

#### Test:
```
TEST_CASE("test # 1", "[tag]") {
    SECTION("something") {
        CHECK(2 == 1);
        SECTION("another thing") {
            CHECK(1 == 2);
        }
    }
}
```

#### Output:
```
test # 1
  something
-------------------------------------------------------------------------------
test/test_the_source.c:15
...............................................................................

test/test_the_source.c:16: FAILED:
  CHECK( 2 == 1 )

-------------------------------------------------------------------------------
test # 1
  something
  another thing
-------------------------------------------------------------------------------
test/test_the_source.c:17
...............................................................................

test/test_the_source.c:18: FAILED:
  CHECK( 1 == 2 )

===============================================================================
test cases: 1 | 1 failed
assertions: 2 | 2 failed
```

* Catch version: **Catch v1.9.3**
* Operating System: **Windows 7 x64 / Windows 10 x64**
* Compiler+version: **g++.exe (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 5.3.0**

**Comment by horenmar** [comment]
I don't see the problem?

If you want to abort test case on failure, don't use `CHECK`, use `REQUIRE`.

**Comment by pJunger** [comment]
Sorry for the bad description.
To me, the current output seems more like two separate paths through the test-case which I would have expected if both Sections would be parallel to one another?

I would have expected to get one result more like this:
```
-------------------------------------------------------------------------------
test # 1
  something
  another thing
-------------------------------------------------------------------------------
test/test_the_source.c:19
...............................................................................

test/test_the_source.c:20: FAILED:
  CHECK( 2 == 1 )

    test/test_the_source.c:21: FAILED:
      CHECK( 1 == 2 )

===============================================================================
test cases: 1 | 1 failed
assertions: 2 | 2 failed
```

**Comment by horenmar** [comment]
That would be an invalid output, because the first failing CHECK isn't in the "another thing" section.

Anyway, the console reporter is a streaming reporter, which means that it works fairly simply -- when an assertion ends, it is notified and takes appropriate action. For console reporter with default settings, this means reporting failing assertion together with its location (which means what test case/sections it is in). Because two assertions fail, two notifications go to stdout, with each one being reporter together with its respective location.


Basically, this is working as intended, but if you want different output, you can always [create your own reporter](/philsquared/Catch/blob/master/docs/reporters.md) and if it is interesting enough, it can become part of Catch's repo.

**Comment by pJunger** [comment]
Okay, I get it now - thanks for your explanation.
